pypose/lietensor/lietensor.py

```diff
@@ -291,9 +309,14 @@ def Mul(self, X, Y):
     def identity(cls, *size, **kwargs):
         return SO3_type.Log(SO3_type.identity(*size, **kwargs))
 
-    def randn(self, *size, sigma=1, requires_grad=False, **kwargs):
-        data = super().randn(*size, sigma=sigma, **kwargs).detach()
-        return LieTensor(data, ltype=so3_type).requires_grad_(requires_grad)
+    def randn(self, *size:_size, sigma:_size_any_t=1.0, requires_grad=False, **kwargs):
+        size = to_tuple(size)
+        data_ = torch.randn(*(size + torch.Size([3])), **kwargs)
+        dis_2 = torch.sum(torch.pow(data_, 2), dim=-1, keepdim=True)
+        dis_ = torch.sqrt(dis_2.to(torch.double)).to(data_.dtype)
```

Comment:
doesn't need to convert to double.

Reply:
use `data.pow(2).sum(dim=-1, keepdim=True)` instead.

Reply:
done